ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m

```diff
@@ -129,62 +101,231 @@ - (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)
-			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+			[self.tableView deleteRowsAtIndexPaths:@[indexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
-			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+			[self.tableView insertRowsAtIndexPaths:@[newIndexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
 			break;
-		default:
-			break;
 	}
 	
 }
 
-- (void) controllerDidChangeContent:(NSFetchedResultsController *)controller {
+- (void)controllerDidChangeContent:(NSFetchedResultsController *)controller {
 	
 	[self.tableView endUpdates];
 	
 }
 
-#pragma mark - DataSource and Delegate
+#pragma mark - Helper methods
+
+- (CGFloat)textViewHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *) tableView {
+	UITextView *calculationView = ((CDAddReminderCell *)[self.tableView cellForRowAtIndexPath:indexPath]).textView;
+	CGFloat textViewWidth = calculationView.frame.size.width;
 	
-	return self.fetchedResultsController.sections.count;
+	if (!calculationView) {
+		return 50;
+	}
 
+	CGSize size = [calculationView sizeThatFits:CGSizeMake(textViewWidth, FLT_MAX)];
+	return size.height;
+	
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+- (void)scrollToCursorForTextView:(UITextView *)textView {
+	
+	CGRect cursorRect = [textView caretRectForPosition:textView.selectedTextRange.start];
+	cursorRect = [self.tableView convertRect:cursorRect fromView:textView];
+	
+	if (![self rectVisible:cursorRect]) {
+		cursorRect.size.height += 8;
+		[self.tableView scrollRectToVisible:cursorRect animated:YES];
+	}
+	
+}
+
+#pragma mark - Keyboard specific methods
+
+- (BOOL)rectVisible:(CGRect)rect {
+	
+	CGRect visibleRect;
+	visibleRect.origin = self.tableView.contentOffset;
+	visibleRect.origin.y += self.tableView.contentInset.top;
+	visibleRect.size = self.tableView.bounds.size;
+	visibleRect.size.height -= self.tableView.contentInset.top + self.tableView.contentInset.bottom;
+	
+	return CGRectContainsRect(visibleRect, rect);
+	
+}
+
+- (void)keyboardWillShow:(NSNotification *)aNotification {
+	
+	NSDictionary* info = [aNotification userInfo];
+	CGSize kbSize = [[info objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue].size;
+	
+	UIEdgeInsets contentInsets = UIEdgeInsetsMake(self.tableView.contentInset.top, 0.0, kbSize.height, 0.0);
+	
+	self.tableView.contentInset = contentInsets;
+	self.tableView.scrollIndicatorInsets = contentInsets;
+	
+}
+
+- (void)keyboardWillHide:(NSNotification*)aNotification {
+	
+	[UIView beginAnimations:nil context:nil];
+	[UIView setAnimationDuration:0.35];
+	
+	UIEdgeInsets contentInsets = UIEdgeInsetsMake(self.tableView.contentInset.top, 0.0, 0.0, 0.0);
+	
+	self.tableView.contentInset = contentInsets;
+	self.tableView.scrollIndicatorInsets = contentInsets;
+	
+	[UIView commitAnimations];
+	
+}
+
+#pragma mark - UITextViewDelegate
```

Comment:
UITextViewDelegate methods should go in the custom cell and be handled there because the textfield is there. To expose actions to the controller you will  implement delegate method for the cell. 

In this way you wont have a large method "+- (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text {" and will group things more logically. 
Remind me of this so we can discuss live, because there are more to describe here and just a comment is not enough.